src/HearThis/Script/ScriptProviderBase.cs

```diff
@@ -220,13 +221,46 @@ public void SetSkippedStyle(string style, bool skipped)
 						details["style"] = style;
 						Analytics.Track("Added skipped style", details);
 						_skippedParagraphStyles.Add(style);
+						BackUpAnyClipsForSkippedStyle(style);
 						Save();
 					}
 				}
 				else
 				{
 					if (_skippedParagraphStyles.Remove(style))
+					{
+						RestoreAnyClipsForUnskippedStyle(style);
 						Save();
+					}
+				}
+			}
+		}
+
+		private void BackUpAnyClipsForSkippedStyle(string style)
+		{
+			// This method will check to see whether the clip exists - does nothing if not.
+			ProcessBlocksHavingStyle(style, ClipRepository.BackUpRecordingForSkippedLine);
+		}
+
+		private void RestoreAnyClipsForUnskippedStyle(string style)
+		{
+			ProcessBlocksHavingStyle(style, (p, b, c, i) => ClipRepository.RestoreBackedUpClip(p, b, c, i));
```

Comment:
RestoreBackedUpClip has a return value (not used in this usage), so its signature doesn't match Action<...>